src/main/kotlin/no/nav/klage/oppgave/clients/kabaldocument/KabalDocumentMapper.kt

```diff
@@ -72,16 +72,25 @@ class KabalDocumentMapper(
             vedlegg.filterIsInstance<JournalfoertDokumentUnderArbeidAsVedlegg>()
                 .sortedByDescending { it.sortKey }
 
-        val datoMottatt = if (hovedDokument.dokumentType == DokumentType.KJENNELSE_FRA_TRYGDERETTEN) {
+        val datoMottatt = if (hovedDokument.isInngaaende()) {
             hovedDokument as OpplastetDokumentUnderArbeidAsHoveddokument
             hovedDokument.datoMottatt
         } else null
 
+        val inngaaendeKanal = when (hovedDokument.dokumentType) {
+            DokumentType.KJENNELSE_FRA_TRYGDERETTEN -> Kanal.ALTINN_INNBOKS
+            DokumentType.ANNEN_INNGAAENDE_POST -> {
+                hovedDokument as OpplastetDokumentUnderArbeidAsHoveddokument
+                Kanal.valueOf(hovedDokument.inngaaendeKanal.toString())
```

Comment:
`.name` i stedet for `.toString()`?

Reply:
Du, denne er spesifikt for å unngå LayeredArchitectureTest. Er en merkelig konsekvens, men bruker den her nå.